_delphi_utils_python/delphi_utils/archive.py

```diff
@@ -98,43 +97,44 @@ def diff_export_csv(
         after_df.loc[added_idx, :])
 
 
-def run_module(archive_type: str,
-               cache_dir: str,
-               export_dir: str,
-               **kwargs):
-    """Build and run an ArchiveDiffer.
+def from_params(params):
+    """Build an ArchiveDiffer from `params`.
+
+    The type of ArchiveDiffer constructed is inferred from the parameters
 
     Parameters
     ----------
-    archive_type: str
-        Type of ArchiveDiffer to run.  Must be one of ["git", "s3"] which correspond to
-        `GitArchiveDiffer` and `S3ArchiveDiffer`, respectively.
-    cache_dir: str
-        The directory for storing most recent archived/uploaded CSVs to start diffing from.
-    export_dir: str
-        The directory with most recent exported CSVs to diff to.
-    **kwargs:
-        Keyword arguments corresponding to constructor arguments for the respective ArchiveDiffers.
+    params: Dict[str, Dict[str, Any]]
+        Dictionary of user-defined parameters with the following structure:
+        - "common":
+            - "export_dir": str, directory to which indicator output files have been exported
+        - "archive":
+            - "cache_dir": str, directory containing cached data from previous indicator runs
+            - "branch_name" (required for git archiver): str, name of git branch
     """
-    if archive_type == "git":
-        arch_diff = GitArchiveDiffer(cache_dir,
-                                     export_dir,
-                                     kwargs["branch_name"],
-                                     kwargs["override_dirty"],
-                                     kwargs["commit_partial_success"],
-                                     kwargs["commit_message"])
-    elif archive_type == "s3":
-        arch_diff = S3ArchiveDiffer(cache_dir,
-                                    export_dir,
-                                    kwargs["bucket_name"],
-                                    kwargs["indicator_prefix"],
-                                    kwargs["aws_credentials"])
-    elif archive_type == "filesystem":
-        arch_diff = FilesystemArchiveDiffer(cache_dir,
-                                            export_dir)
-    else:
-        raise ValueError(f"No archive type named '{archive_type}'")
-    arch_diff.run()
+    if "archive" not in params:
+        return None
+
+    common_params = params["common"]
+    export_dir = common_params["export_dir"]
```

Comment:
```suggestion
    export_dir = common_params["common"]["export_dir"]
```

Don't feel strongly about this recommendation, seems like extracting the sub-dicts to their own variable is done a few times here. Also looks like this is mainly copy paste.

Reply:
You are actually suggesting to change this to `export_dir = params["common"]["export_dir"]`, yes?

Reply:
yea

Reply:
This is obsolete now that I have changed the logic